File: fixtures/python/redos_regex/positive.py
"""Positive fixture for SEC014-redos-regex — patterns that cause catastrophic backtracking."""
import re

# Nested repetition: (a+)+ is the canonical ReDoS pattern.
pattern1 = re.compile(r"(a+)+")

# Nested repetition: (.*)*
pattern2 = re.compile(r"(.*)*")

# Nested repetition: (\w+)+ — common in user-input validators
pattern3 = re.compile(r"(\w+)+end")

File: fixtures/rust/redos_regex/positive.rs
// Positive fixture for SEC014-redos-regex — patterns that cause catastrophic backtracking.
use regex::Regex;

fn catastrophic_patterns() {
    // Nested repetition: (a+)+ is the canonical ReDoS pattern.
    let _r1 = Regex::new("(a+)+").unwrap();

    // Nested repetition: (.*)*
    let _r2 = Regex::new("(.*)*").unwrap();
}
